lib/storage/index.ts

```diff
@@ -1,13 +1,52 @@
+import * as Logger from '../Logger';
+
 import PlatformStorage from './platforms';
 import InstanceSync from './InstanceSync';
+import NoopProvider from './providers/NoopProvider';
 import type StorageProvider from './providers/types';
 
-const provider = PlatformStorage;
+let provider = PlatformStorage;
 let shouldKeepInstancesSync = false;
+let finishInitalization: (value?: unknown) => void;
+const initPromise = new Promise((resolve) => {
+    finishInitalization = resolve;
+});
 
 type Storage = {
     getStorageProvider: () => StorageProvider;
-} & StorageProvider;
+} & Omit<StorageProvider, 'name'>;
+
+/**
+ * Degrade performance by removing the storage provider and only using cache
+ */
+function degradePerformance(error: Error) {
+    Logger.logAlert(`Error while using ${provider.name}. Falling back to only using cache and dropping storage.`);
```

Comment:
Just to confirm, starting from here, the app won't actually update anything based on what the user received, and the data will always be stale, right?

Reply:
> Just to confirm, starting from here, the app won't actually update anything based on what the user received, and the data will always be stale, right?

@danieldoglas Yes, in case of error (out-of-memory) we will not store anything and the data will be always stale